packages/react-components/src/components/Checkbox/Checkbox.module.scss

```diff
@@ -21,44 +28,28 @@
     background-color: var(--surface-primary-default);
     width: 16px;
     height: 16px;
-  }
-
-  &__checkmark {
-    transition: all 0.2s ease;
-    opacity: 0;
-    pointer-events: none;
-  }
 
-  &--selected {
-    &:hover {
-      .checkbox__input:checked + .checkbox__square {
-        border-color: var(--action-primary-hover);
-        background-color: var(--action-primary-hover);
-      }
+    &::after {
+      display: inline-block;
+      position: absolute;
+      transition: opacity 0.2 ease;
+      opacity: 0;
+      z-index: 1;
+      background-color: var(--color-white);
+      width: 24px;
+      height: 24px;
+      content: '';
+      clip-path: path(
+        'M9 16.2001L5.5 12.7001C5.11 12.3101 4.49 12.3101 4.1 12.7001C3.71 13.0901 3.71 13.7101 4.1 14.1001L8.29 18.2901C8.68 18.6801 9.31 18.6801 9.7 18.2901L20.3 7.70007C20.69 7.31007 20.69 6.69007 20.3 6.30007C19.91 5.91007 19.29 5.91007 18.9 6.30007L9 16.2001Z'
+      );
+      scale: 0.5;
```

Comment:
It won't be easier to change the background color of the input and just show the icon (positioned by CSS) instead of a that pseudo element? 🤔

Reply:
How would you see that? I think having one less thing to worry about in the markup is a better approach. I suppose both the Checkbox and the icon could be siblings, positioned absolutely on top of each other, but this solution feels better to me, WDYT?

Reply:
Yes, I agree that one less thing in markup will be better.
How about to use the checkmark icon directly in the `content` of that pseudo-element?
`content: url([path]/icon.svg);`

Reply:
I wanted to do that in the first place, would you be able to help me out? I'm not sure where to point the path

Reply:
It could be something like this `content: url('./check.svg');` and icon file should be placed in the same place where we keep the styles for this component. Of course, the icon should have the correct color in every mode, so we must change the `fill` or `stroke` param to token: `--content-invert-primary`.